### ISSUE TITLE
DeprecationWarning: 'U' mode is deprecated

### ISSUE THREAD

**Issue by henrykironde** [weecology/retriever#1223]
pytest-dev/py it's core base has some changes. 
And we have a warning as below

```bash

test/test_integration.py::test
  c:\python35\lib\site-packages\pymysql\cursors.py:170: Warning: (1051, "Unknown table 'testdb.data_no_header'")
    result = self._query(query)
  c:\python35\lib\site-packages\retriever\lib\engine_tools.py:295: DeprecationWarning: 'U' mode is deprecated
    input_obj = io.open(input_file, 'rU')

```
**The options based on usefulness:**
1.  Revisit any line that has ` io.open(arg, 'rU')`  and probably change to   ` io.open(arg, 'r')`

**Or but not recommended for long term:**

2. Find a way to temporarily suppress the warnings with -W ignore::DeprecationWarning or```
 import warnings
warnings.filterwarnings("ignore", category=DeprecationWarning) 
```

ref: https://github.com/pytest-dev/py/pull/116

**Comment by amanjain25** [comment]
I would like to work on this

**Comment by ethanwhite** [comment]
Fixed by 4d1341d0c11ba32d4d009b03adea29d6baa006d2. Thanks @harshitbansal05!
